libs/cosmos-sdk/x/capability/keeper/keeper_test.go

```diff
@@ -0,0 +1,292 @@
+		mods, gotCap, err := sk.LookupModules(suite.ctx, "transfer")
+
+		suite.Require().True(ok, "could not retrieve owners")
+		suite.Require().NotNil(owners, "owners is nil")
+
+		suite.Require().NoError(err, "could not retrieve modules")
+		suite.Require().NotNil(gotCap, "capability is nil")
+		suite.Require().NotNil(mods, "modules is nil")
+		suite.Require().Equal(cap, gotCap, "caps not equal")
+
+		suite.Require().Equal(len(expectedOrder), len(owners.Owners), "length of owners is unexpected")
+		for i, o := range owners.Owners {
+			// Require owner is in expected position
+			suite.Require().Equal(expectedOrder[i], o.Module, "module is unexpected")
+			suite.Require().Equal(expectedOrder[i], mods[i], "module in lookup is unexpected")
+		}
+	}
+
+	// foo module releases capability
+	err = sk3.ReleaseCapability(suite.ctx, cap)
+	suite.Require().Nil(err, "could not release capability")
+
+	// new expected order and scoped capabilities
+	expectedOrder = []string{"bank", stakingtypes.ModuleName}
+	sks = []keeper.ScopedKeeper{sk1, sk2}
+
+	// Ensure all scoped keepers can get owners
+	for _, sk := range sks {
+		owners, ok := sk.GetOwners(suite.ctx, "transfer")
+		mods, cap, err := sk.LookupModules(suite.ctx, "transfer")
+
+		suite.Require().True(ok, "could not retrieve owners")
+		suite.Require().NotNil(owners, "owners is nil")
+
+		suite.Require().NoError(err, "could not retrieve modules")
+		suite.Require().NotNil(cap, "capability is nil")
+		suite.Require().NotNil(mods, "modules is nil")
+
+		suite.Require().Equal(len(expectedOrder), len(owners.Owners), "length of owners is unexpected")
+		for i, o := range owners.Owners {
+			// Require owner is in expected position
+			suite.Require().Equal(expectedOrder[i], o.Module, "module is unexpected")
+			suite.Require().Equal(expectedOrder[i], mods[i], "module in lookup is unexpected")
+		}
+	}
+
+	_, ok := sk1.GetOwners(suite.ctx, "  ")
+	suite.Require().False(ok, "got owners from empty capability name")
+}
+
+func (suite *KeeperTestSuite) TestReleaseCapability() {
+	sk1 := suite.keeper.ScopeToModule("bank")
+	sk2 := suite.keeper.ScopeToModule(stakingtypes.ModuleName)
+
+	cap1, err := sk1.NewCapability(suite.ctx, "transfer")
+	suite.Require().NoError(err)
+	suite.Require().NotNil(cap1)
+
+	suite.Require().NoError(sk2.ClaimCapability(suite.ctx, cap1, "transfer"))
+
+	cap2, err := sk2.NewCapability(suite.ctx, "bond")
+	suite.Require().NoError(err)
+	suite.Require().NotNil(cap2)
+
+	suite.Require().Error(sk1.ReleaseCapability(suite.ctx, cap2))
+
+	suite.Require().NoError(sk2.ReleaseCapability(suite.ctx, cap1))
+	got, ok := sk2.GetCapability(suite.ctx, "transfer")
+	suite.Require().False(ok)
+	suite.Require().Nil(got)
+
+	suite.Require().NoError(sk1.ReleaseCapability(suite.ctx, cap1))
+	got, ok = sk1.GetCapability(suite.ctx, "transfer")
+	suite.Require().False(ok)
+	suite.Require().Nil(got)
+
+	suite.Require().Error(sk1.ReleaseCapability(suite.ctx, nil))
```

Comment:
need to check prefixStore's indexkey when update

Reply:
![image](https://user-images.githubusercontent.com/6890141/203957532-e7253957-2495-429f-a87c-7090a59dfa01.png)
when firstly call release ,2nd GetCapablily mean the same with check prefixStore's indexkey when update